### PR TITLE
feat: add number of search results and highlight query term

### PR DESCRIPTION
**Pull request by minz1027**
- Show number of search results
- Highlight query term

screen shot 2016-12-15 at 9 58 30 am

Related to https://github.com/screwdriver-cd/screwdriver/issues/394

### CODE REVIEW

**Comment by tkyi** [comment]
- Would be nice if there was some search term history information (if something was misspelled)
- Also some more metadata for search results (pipeline admin?, date created?)

File: app/components/search-list/template.hbs [hunk 1]
```diff
@@ -1,11 +1,15 @@
+<div class="num-results">
+{{if isEmpty "No Results" (concat "Found " filteredPipelines.length " results") }}
```
Comment (minz1027) [inline app/components/search-list/template.hbs]: Using `result(s)`! yay!

File: app/components/search-list/styles.scss [hunk 2]
```diff
@@ -21,18 +21,24 @@ a {
   color: $sd-link;
   padding: 10px;
   border-bottom: 1px solid $sd-separator;
+}
 
-  .appId {
-    display: inline-block;
-    width: 400px;
-  }
+.appId {
```
Comment (tkyi) [inline app/components/search-list/styles.scss]: Should this still be nested under `a`? Or at least be `a .appId {`?

File: app/components/search-list/styles.scss [hunk 3]
```diff
@@ -21,18 +21,24 @@ a {
...
+  width: 400px;
+}
 
-  .branch {
-    display: inline-block;
-    width: 215px;
-    text-align: right;
-    color: $sd-text;
-  }
+.branch {
```
Comment (tkyi) [inline app/components/search-list/styles.scss]: same here

File: app/components/search-list/styles.scss (hunk 2 above)
Comment (petey) [inline app/components/search-list/styles.scss]: No, this class applies to a span.

File: app/components/search-list/styles.scss (hunk 3 above)
Comment (petey) [inline app/components/search-list/styles.scss]: This also applies to a span.

File: app/components/search-list/template.hbs [hunk 4]
```diff
@@ -1,11 +1,21 @@
+<div class="num-results">
+  {{if isEmpty "No Results" (concat "Found " filteredPipelines.length " result(s)") }}
+</div>
+<div class="header">
```
Comment (petey) [inline app/components/search-list/template.hbs]: This is a good place to use a table. Would greatly simplify your css, too.

Comment (minz1027) [inline app/components/search-list/template.hbs]: @petey should we use table for just the header or both header and the records?

File: app/components/search-list/template.hbs [hunk 5]
```diff
@@ -1,11 +1,21 @@
...
+  <span class="heading admin">Admin</span>
+</div>
 <ul>
   {{#each filteredPipelines as |pipeline|}}
   <li>
+    {{#highlight-terms query}}
     {{#link-to "pipeline" pipeline.id}}
       <span class="appId">{{pipeline.appId}}</span>
       <span class="branch">{{fa-icon "fa-code-fork"}}{{pipeline.branch}}</span>
+      <span class="admin">{{pipeline.admin}}</span>
```
Comment (stjohnjohnson) [inline app/components/search-list/template.hbs]: Please don't list admins, it's really not useful information on the search page.

File: app/components/search-list/template.hbs (hunk 4 above)
Comment (minz1027) [inline app/components/search-list/template.hbs]: fixed

File: app/components/search-list/template.hbs (hunk 5 above)
Comment (minz1027) [inline app/components/search-list/template.hbs]: fixed

Comment (d2lam) [inline app/components/search-list/template.hbs]: 😿

**Pull request merged**